### DISCUSSION TITLE
How do I configure my Flyte deployment to include specialized behavior per project or domain?

### DISCUSSION THREAD

**Discussion by SandraGH5** [flyteorg/flyte#1069] [Q&A] [answered]

**Comment by SandraGH5** [comment, marked as answer]
Please visit "Configuring Your Flyte Deployment" at https://docs.flyte.org/projects/cookbook/en/latest/auto/deployment/cluster/config_flyte_deploy.html.

**Reply by palchicz** [reply]
I think this link is broken. I am also curious about this. I'm wondering if one can set certain environment variables depending on the domain

**Reply by SandraGH5** [reply]
Hi Zach,
I believe this is the updated link:
https://docs.flyte.org/en/latest/deployment/cluster_config/general.html#deployment-cluster-config-general

**Reply by katrogan** [reply]
hey @palchicz could you clarify your use case a little more, do you want to change environment variables in container definitions for tasks by project or domain?

**Reply by palchicz** [reply]
> do you want to change environment variables in container definitions for tasks by project or domain 

That's right, but I can provide a little more background. One could imagine that you have a development database and a production database, each with their own uri. In the development domain, you'd want some task to hit the development database. However, in the production domain, you'd want the same task to hit the production database.

One option is to use environment variables set on the container (differing by domain) to inject the appropriate uri into the task... but that's just one idea. I'm open to other thoughts if the flyte crew has a recommended best practice.

**Reply by katrogan** [reply]
hey @palchicz in this case I think the cluster resource controller stuff might not be the right level of customization :) This is used to customize namespace attributes on your kubernetes cluster, things like resource quotas, etc.

I think the env variable behavior is your best bet. For others reading, at runtime, flytekit tasks can reference `FLYTE_INTERNAL_EXECUTION_DOMAIN` as an environment variable, which you could use to compose the correct database URI in code
